index.d.ts

```diff
@@ -17,6 +17,8 @@ declare namespace pSettle {
 	type PromiseResult<ValueType> = pReflect.PromiseResult<ValueType>;
 	type PromiseFulfilledResult<ValueType> = pReflect.PromiseFulfilledResult<ValueType>;
 	type PromiseRejectedResult = pReflect.PromiseRejectedResult;
+
+	type Awaited<T> = T extends undefined ? T : T extends PromiseLike<infer U> ? U : T;
```

Comment:
Can you add a TODO comment to remove this when TS supports this natively (which it will in TS 4).

Reply:
Done